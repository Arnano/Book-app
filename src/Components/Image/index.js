import React from 'react';
import PropTypes from 'prop-types';
import StyledImage, { StyledImageLink } from './style';

const Image = ({ alt, classes, src }) =>
  <StyledImage
    alt={alt}
    className={classes} 
    src={src} />

const ImageLink = ({ alt, classes, link, src }) => 
  <StyledImageLink href={link} >
      <StyledImage
        alt={alt}
        className={classes} 
        src={src} />
  </StyledImageLink>

const ImageLinkNewWindow = ({ alt, classes, link, rel, src, target }) => 
  <StyledImageLink 
    href={link}
    rel={rel}
    target={target} >
      <StyledImage
        alt={alt}
        className={classes} 
        src={src} />
  </StyledImageLink>

Image.propTypes = {
  alt: PropTypes.string.isRequired,
  classes: PropTypes.string,
  src: PropTypes.string.isRequired
}

ImageLink.propTypes = {
  alt: PropTypes.string.isRequired,
  classes: PropTypes.string,
  link: PropTypes.string.isRequired,
  src: PropTypes.string.isRequired
}

ImageLinkNewWindow.defaultProps = {
  rel: 'noopener noreferrer',
  target: '_blank'
}

ImageLinkNewWindow.propTypes = {
  alt: PropTypes.string.isRequired,
  classes: PropTypes.string,
  link: PropTypes.string.isRequired,
  rel: PropTypes.string,
  src: PropTypes.string.isRequired,
  target: PropTypes.string
}

export default Image;
export { ImageLink, ImageLinkNewWindow };
