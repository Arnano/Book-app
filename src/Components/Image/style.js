import Styled from 'styled-components';

const StyledImage = Styled.img `
  width: 100%;
  height: 300px;
`;

const StyledImageLink = Styled.a `
  text-decoration: none;
  underline: none;
  cursor: pointer;

  & img:hover {
    opacity: 0.8;
    transition: opacity 0.3s ease;
  }
`;

export default StyledImage;
export { StyledImageLink };
