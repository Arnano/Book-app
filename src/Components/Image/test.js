import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Image, { ImageLink } from './';

Enzyme.configure({
  adapter: new Adapter()
}); 

describe('Image', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Image src={'http://via.placeholder.com/350x150'} alt={'Placeholder'}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('has a valid snapshot', () => {
    const component = renderer.create(<Image src={'http://via.placeholder.com/350x150'} alt={'Placeholder'}/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Image link', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ImageLink link={'https://google.com'} src={'http://via.placeholder.com/350x150'} alt={'Placeholder'}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('has a valid snapshot', () => {
    const component = renderer.create(<ImageLink link={'https://google.com'} src={'http://via.placeholder.com/350x150'} alt={'Placeholder'}/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
