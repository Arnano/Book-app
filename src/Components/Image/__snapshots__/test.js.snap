// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image has a valid snapshot 1`] = `
<img
  alt="Placeholder"
  className="sc-bdVaJa bKjKWq"
  src="http://via.placeholder.com/350x150"
/>
`;

exports[`Image link has a valid snapshot 1`] = `
<a
  className="sc-bwzfXH kihqCD"
  href="https://google.com"
>
  <img
    alt="Placeholder"
    className="sc-bdVaJa bKjKWq"
    src="http://via.placeholder.com/350x150"
  />
</a>
`;
